strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/configuration/StoragesConfigurationController.java

```diff
@@ -227,31 +227,40 @@ public ResponseEntity addOrUpdateRepository(@ApiParam(value = "The storageId", r
                                                 BindingResult bindingResult,
                                                 @RequestHeader(HttpHeaders.ACCEPT) String accept)
     {
-        if (bindingResult.hasErrors())
+        
+        if(configurationManagementService.getConfiguration().getStorage(storageId) != null) 
         {
-            throw new RequestBodyValidationException(FAILED_SAVE_REPOSITORY, bindingResult);
-        }
-        try
-        {
-            MutableRepository repository = conversionService.convert(repositoryForm, MutableRepository.class);
-
-            logger.debug("Creating repository " + storageId + ":" + repositoryId + "...");
-
-            configurationManagementService.saveRepository(storageId, repository);
-
-            final RepositoryPath repositoryPath = repositoryPathResolver.resolve(new Repository(repository));
-            if (!Files.exists(repositoryPath))
+        	if (bindingResult.hasErrors())
```

Comment:
Cosmetic: The indentation of this `if`-`else` block is not according to our coding convention. Would you mind correcting this? Thanks! :)